src/crypto/noise/mod.rs

```diff
@@ -113,85 +113,71 @@ impl NoiseContext {
-            NOISE_PARAMETERS.parse().expect("valid Noise pattern"),
+            NOISE_PARAMETERS.parse().expect("qed; Valid noise pattern"),
             Box::new(protocol::Resolver),
         );
 
-        let dh_keypair = builder.generate_keypair().expect("keypair generation to succeed");
+        let dh_keypair = builder.generate_keypair()?;
         let static_key = &dh_keypair.private;
 
         let noise = match role {
-            Role::Dialer => builder
-                .local_private_key(static_key)
-                .build_initiator()
-                .expect("initialization to succeed"),
-            Role::Listener => builder
-                .local_private_key(static_key)
-                .build_responder()
-                .expect("initialization to succeed"),
+            Role::Dialer => builder.local_private_key(static_key).build_initiator()?,
+            Role::Listener => builder.local_private_key(static_key).build_responder()?,
         };
 
         Self::assemble(noise, dh_keypair, keypair, role)
     }
 
     /// Create new [`NoiseContext`] with prologue.
-    pub fn with_prologue(id_keys: &Keypair, prologue: Vec<u8>) -> Self {
+    pub fn with_prologue(id_keys: &Keypair, prologue: Vec<u8>) -> crate::Result<Self> {
         let noise: Builder<'_> = Builder::with_resolver(
-            NOISE_PARAMETERS.parse().expect("valid Noise pattern"),
+            NOISE_PARAMETERS.parse().expect("qed; Valid noise pattern"),
             Box::new(protocol::Resolver),
         );
 
-        // let noise = snow::Builder::new(NOISE_PARAMETERS.parse().expect("valid Noise patterns"));
-        let keypair = noise.generate_keypair().unwrap();
+        let keypair = noise.generate_keypair()?;
 
         let noise = noise
             .local_private_key(&keypair.private)
             .prologue(&prologue)
-            .build_initiator()
-            .expect("to succeed");
+            .build_initiator()?;
 
         Self::assemble(noise, keypair, id_keys, Role::Dialer)
     }
 
     /// Get remote public key from the received Noise payload.
-    // TODO: refactor
     pub fn get_remote_public_key(&mut self, reply: &[u8]) -> crate::Result<PublicKey> {
-        if reply.len() <= 2 {
-            return Err(error::Error::InvalidData);
-        }
-
-        // TODO: no unwraps
-        let size: Result<[u8; 2], _> = reply[0..2].try_into();
-        let _size = u16::from_be_bytes(size.unwrap());
+        let (len_slice, reply) = reply.split_at(2);
+        let len = u16::from_be_bytes(len_slice.try_into().map_err(|_| error::Error::InvalidData)?)
+            as usize;
 
-        // TODO: buffer size
-        let mut inner = vec![0u8; 1024];
+        let mut buffer = vec![0u8; len];
 
         let NoiseState::Handshake(ref mut noise) = self.noise else {
-            panic!("invalid state to read the second handshake message");
+            tracing::error!(target: LOG_TARGET, "invalid state to read the second handshake message");
```

Comment:
May be add a `debug_assert!` after this line?

```suggestion
            tracing::error!(target: LOG_TARGET, "invalid state to read the second handshake message");
            debug_assert!(false);
```